metagpt/provider/openai_api.py

```diff
@@ -209,7 +234,14 @@ def get_choice_function_arguments(self, rsp: ChatCompletion) -> dict:
             and message.tool_calls[0].function.arguments is not None
         ):
             # reponse is code
-            return json.loads(message.tool_calls[0].function.arguments, strict=False)
+            try:
+                return json.loads(message.tool_calls[0].function.arguments, strict=False)
+            except json.decoder.JSONDecodeError as e:
+                error_msg = (
+                    f"Got JSONDecodeError for \n{'--'*40} \n{message.tool_calls[0].function.arguments}, {str(e)}"
+                )
+                logger.error(error_msg)
+                return self._parse_arguments(message.tool_calls[0].function.arguments)
```

Comment:
I don't think that's a good approach. We should either reuse the repair-related method, or we should retry until it succeeds

Reply:
Retry until it succeeds sounds good.

Reply:
I try it like this, but it does not work.
![image](https://github.com/geekan/MetaGPT/assets/84325228/ccf53644-b462-42df-9cf9-279b703df44e)
![image](https://github.com/geekan/MetaGPT/assets/84325228/699c620a-22fa-4c28-9aa0-92a0bc1c3157)

And,  I found these in openai functions:
![image](https://github.com/geekan/MetaGPT/assets/84325228/f2dd2c4e-b929-4a1a-b0af-b248e391694e)
When parsing the results of function calls, it seems necessary to use regular expressions for matching.